text/trace/0168-sampling-propagation.md

```diff
@@ -0,0 +1,181 @@
+Traces](https://arxiv.org/pdf/2107.07703.pdf) section 2.8 for a
+detailed explanation.
+
+Such a random variable `r` can be generated using the following
+pseudocode:
+
+```
+r := 0
+for {
+  if nextRandomBit() {
+    break // The expected value of r is 2
+  }
+  r++
+}
+```
+
+This can be computed from a stream of random bits as the number of
+leading zeros using efficient instructions on modern computer
+architectures.
+
+For example, the value 3 means there were three leading zeros and
+corresponds with being sampled at probabilities 1-in-1 through 1-in-8
+but not at probabilities 1-in-16 and smaller.  Using one byte of
+information we can convey a consistent sampling decision for sampling
+rates as small as 2^-255.
+
+### Proposed `tracestate` syntax
+
+The consistent sampling decision and head trace sampling probability
+will be propagated using four bytes of base16 content, as follows:
+
+```
+tracestate: otelprob=PPRR
+```
+
+where `PP` are two bytes of base16 probability value and `RR` are two
+bytes of base16 random value.
+
+### Examples
+
+The following `tracestate` value:
+
+```
+tracestate: otelprob=0a03
+```
+
+translates to
+
+```
+base16(probability) = 03 // 1-in-8 head probability
+base16(randomness) = 0a // qualifies for 1-in-1024 sampling or greater
+```
+
+Any `TraceIDRatioBased` Sampler configured with probability 2^-10 or
+greater will enable sampling this trace, whereas any
+`TraceIDRatioBased` Sampler configured with probability 2^-11 or less
+will stop sampling this trace.  The W3C `sampled` flag is set to true
+when the probability value is less than or equal to the randomness
+value.
+
+## Internal details
+
+The reasoning behind restricting the set of sampling rates is that it:
+
+- Lowers the cost of propagating head sampling probability
+- Limits the number of random bits required
+- Makes math involving partial traces tractable.
+
+[An algorithm for making statistical inference from partially-sampled
+traces has been published](https://arxiv.org/pdf/2107.07703.pdf) that
+explains how to work with a limited number of power-of-2 sampling rates.
+
+### Behavior of the `TraceIDRatioBased` Sampler
+
+The Sampler must be configured with a power-of-two probability
+`P=2^-S`.
+
+Using one byte to represent both the probability and randomness values
+means each value is limited to 255.  As a special case, the head
+probability `P=0` is represented using the probability value `S=255`,
+meaning `P=0` is indistinguishable from `P=2^-255`.  We propose to
+limit valid head sampling probabilities to `P=2^-254` or greater to
+address this ambiguity.
+
+If the context is a new root, the initial `tracestate` must be created
+using geometrically-distributed random value `R` (as described above,
+with maximum value 254) and the initial head probability value `S`.
+
+If the context is not a new root, output a new `tracestate` with the
+same `R` value as the parent context, using the Sampler's own value of
+`S` for the head probability.
```

Comment:
Just to be clear, I am propagating tracestate with my own S value, not the value I received from the parent?

Reply:
Yes, this is what I mean by "the Sampler's own value of `S`". I'll try to improve this text.

Reply:
I've improved this text.